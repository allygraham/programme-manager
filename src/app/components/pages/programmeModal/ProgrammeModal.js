import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import { Modal } from '../../views/modal';
import { Form } from '../../views/form';
import { constants } from '../../../constants';

const Title = styled('h2')`
  margin-bottom: 1rem;
  font-size: 1.5rem;
  font-weight: 700;
`;

export const ProgrammeModal = ({ onSubmit, formData, isEditing }) => (
  <Modal>
    <Title>{isEditing ? constants.modalTitles.edit : constants.modalTitles.add }</Title>
    <Form
      onSubmit={onSubmit}
      formData={formData}
      isEditing={isEditing}
    />
  </Modal>
);

ProgrammeModal.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  formData: PropTypes.object.isRequired,
  isEditing: PropTypes.bool.isRequired,
};
