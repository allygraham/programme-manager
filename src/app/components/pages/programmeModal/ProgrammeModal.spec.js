import React from 'react';
import { shallow } from 'enzyme';

import { ProgrammeModal } from './ProgrammeModal';

describe('ProgrammeModal', () => {

  test('render the <ProgrammeModal> component', () => {
    const formData = { id: 123, name: 'The Chase' };
    const onSubmit = jest.fn();
    const isEditing = false;

    const wrapper = shallow(<ProgrammeModal
      formData={formData}
      onSubmit={onSubmit}
      isEditing={isEditing}
    />);
    expect(wrapper).toMatchSnapshot();
  });

});
