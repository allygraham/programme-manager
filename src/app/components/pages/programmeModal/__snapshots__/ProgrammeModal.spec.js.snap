// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgrammeModal render the <ProgrammeModal> component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgrammeModal
    formData={
      Object {
        "id": 123,
        "name": "The Chase",
      }
    }
    isEditing={false}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(h2)>
          Add Programme
        </Styled(h2)>,
        <Form
          formData={
            Object {
              "id": 123,
              "name": "The Chase",
            }
          }
          isEditing={false}
          onSubmit={[MockFunction]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Add Programme",
        },
        "ref": null,
        "rendered": "Add Programme",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "formData": Object {
            "id": 123,
            "name": "The Chase",
          },
          "isEditing": false,
          "onSubmit": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(h2)>
            Add Programme
          </Styled(h2)>,
          <Form
            formData={
              Object {
                "id": 123,
                "name": "The Chase",
              }
            }
            isEditing={false}
            onSubmit={[MockFunction]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Add Programme",
          },
          "ref": null,
          "rendered": "Add Programme",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "formData": Object {
              "id": 123,
              "name": "The Chase",
            },
            "isEditing": false,
            "onSubmit": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
